packages/block-library/src/code/save.js

```diff
@@ -13,7 +13,7 @@ export default function save( { attributes } ) {
 		<pre { ...useBlockProps.save() }>
 			<RichText.Content
 				tagName="code"
-				value={ escape( attributes.content ) }
+				value={ escape( attributes.content.toString() ) }
```

Comment:
could we not skip the `escape()` call here since `RichText.Content` could perform that internally and for every case? is there ever a time where we wouldn't want to `escape` it?

`RichText.Content` knows how to safely convert a `RichTextValue` into safe HTML, and it knows how to turn a string into a `RichTextValue`

Reply:
I think the escaping here is for shortcode brackets.

Reply:
Yep, it's encoding short codes and links to prevent them from being replaced by display filters.

I wonder if this is the right approach though, shouldn't this be a concern of the display filter to check the context?

https://github.com/WordPress/gutenberg/blob/6d9740f13f8232958b4c0e48fb120c189b216278/packages/block-library/src/code/utils.js#L12-L17

Reply:
For URLs, I don't understand the reasoning at all. URLs in blocks should never embed. We embed them automatically in the editor to a dedicated block. The auto embedding should only happen for classic content or the embed block.

Shortcodes is a bit trickier because they can appear inline, but still, just check if it's in a code tag? Maybe with your tag processor? 😉